br/pkg/backup/limit.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backup
+
+import "sync"
+
+type ResourceConcurrentLimiter struct {
+	cond      *sync.Cond
+	current   int
+	threshold int
+}
+
+func NewResourceMemoryLimiter(threshold int) *ResourceConcurrentLimiter {
```

Comment:
do we have to make a utility class here or we can reuse some existing util instead?

Reply:
It seems it is only used here in TiDB repository. And there is no the same struct to limit resource.

Reply:
Therefore, I put it in the `backup` package as its internal function.